boogiestats/boogie_ui/views.py

```diff
@@ -238,8 +238,14 @@ def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
 
         song_hash = self.kwargs["song_hash"]
-        context["song"] = Song.get_or_404(hash=song_hash)
-        context["num_highscores"] = Score.objects.filter(song_id=song_hash, is_top=True).count()
+        song = Song.get_or_404(hash=song_hash)
+        context["song"] = song
+        context["num_highscores"] = song.scores.filter(is_top=True).count()
+        if hasattr(self.request.user, "player"):
+            context["my_scores"] = song.scores.filter(player=self.request.user.player).count()
+
+        if "player_id" in self.kwargs:
+            context["player"] = Player.get_or_404(id=self.kwargs["player_id"])
```

Comment:
there's a new syntax since python 3.8 that allows writing this in a slightly cleaner way, it's called ["walrus operator": `:=`](https://docs.python.org/3/whatsnew/3.8.html#assignment-expressions). If we used it, this could look like:

```python
        if player_id := self.kwargs.get("player_id"):
            context["player"] = Player.get_or_404(id=player_id)
```

Reply:
oh that's a cool syntax, I didn't know about it
I changed said lines to this new syntax